org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WStringEditor.java

```diff
@@ -206,6 +208,36 @@ else if (gridField.getDisplayType() == DisplayType.Memo)
 	@Override
 	public void onEvent(Event event)
     {
+		if (Events.ON_OK.equals(event.getName())) {
+		    if (event instanceof KeyEvent) {
+		        KeyEvent keyEvent = (KeyEvent) event;
+		        if (keyEvent.isShiftKey() && getComponent().isMultiline() && tableEditor) {
+		            Component target = event.getTarget();
+		            if (target instanceof Textbox) {
+		                String uuid = target.getUuid();
+
+		                String script = String.format(
+		                	    "(function() {" +
+		                	    "  var elem = jq('#%s')[0];" +
+		                	    "  if (!elem) return;" +
+		                	    "  var start = elem.selectionStart;" +
+		                	    "  var end = elem.selectionEnd;" +
+		                	    "  var value = elem.value;" +
+		                	    "  elem.value = value.substring(0, start) + '\\n' + value.substring(end);" +
+		                	    "  elem.selectionStart = elem.selectionEnd = start + 1;" +
+		                	    "  var cmp = zk.Widget.$('$%s');" +
+		                	    "  if (cmp) cmp.fire('onChanging', {value: elem.value});" +
```

Comment:
should get widget at the start and use $n() (see https://www.zkoss.org/javadoc/7.0.3/jsdoc/zk/Widget.html) to get the dom element. Also, usually, we use zk.Widget.$('#uuid') instead. Another thing - it is using selectionStart and selectionEnd, wondering whether this would works correctly if user make real text selection before pressing shift+enter.

Reply:
Thanks @hengsin, just updated the script with your suggestions. 

> Another thing - it is using selectionStart and selectionEnd, wondering whether this would works correctly if user make real text selection before pressing shift+enter.

Yes - this is to behave exactly how it does natively on Form View, when the user selects a text and hit Shift+Enter, the selected text is replaced by a newline.

Please let me know if more adjustments to the script should be done